src/Microsoft.IdentityModel.Abstractions/TelemetryEventDetails.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Details of the telemetry event.
+    /// </summary>
+    /// <remarks>
+    /// This implementation is not meant to be thread-safe. This implementation would either need to be overridden or
+    /// usage should not be concurrently operated on.
+    /// </remarks>
+    public abstract class TelemetryEventDetails
+    {
+        /// <summary>
+        /// The underlying properties making up the <see cref="TelemetryEventDetails"/>.
+        /// </summary>
+        protected internal Dictionary<string, object> PropertyValues { get; } = new Dictionary<string, object>();
```

Comment:
This should return 'IDictionary' rather than 'Dictionary'.

Reply:
Good catch